lib/cooked_post_processor.rb

```diff
@@ -330,7 +330,15 @@ def convert_to_link!(img)
       img["height"] = height
     end
 
-    if upload = Upload.get_from_url(src)
+    upload = Upload.get_from_url(src)
+
+    # if the upload already exists and is attached to a different post,
+    # or the original_sha1 is missing meaning it was created before secure
+    # media was enabled. we want to re-thumbnail and re-optimize in this case
+    # to avoid using uploads linked to many other posts
+    upload = Upload.invalid_secure_upload_reuse?(upload, @post) ? nil : upload
```

Comment:
It seems in both places in the PR where this new function is used, you also use a ternary operator to consider changing the upload to nil. 

Why not have the method return an `upload` or `nil`? Something like:

`upload = Upload.consider_for_reuse(upload, @post)`

Reply:
That is very true...will change!